pkg/controller/periodicity/periodicity_controller.go

```diff
@@ -94,29 +86,35 @@ func (c *Controller) syncStatefulSetTimeStamp() error {
 	if err != nil {
 		return err
 	}
+
 	var errs []error
 	for _, sts := range stsList {
 		// If there is any error during our sts annotation updating, we just collect the error
 		// and continue to next sts
-		ok, tcRef := util.IsOwnedByTidbCluster(sts)
+		ok, _ := util.IsOwnedByTidbCluster(sts)
 		if !ok {
 			continue
 		}
-		tc, err := c.tcLister.TidbClusters(sts.Namespace).Get(tcRef.Name)
-		if err != nil {
-			errs = append(errs, err)
-			continue
-		}
 		if sts.Annotations == nil {
 			sts.Annotations = map[string]string{}
 		}
 		sts.Annotations[label.AnnStsLastSyncTimestamp] = time.Now().Format(time.RFC3339)
```

Comment:
create a new annotations map, otherwise, cache in informer will be modified.

Reply:
updated.